### PR TITLE
Add support for Oracle Cloud

### PR DESCRIPTION
**Pull request by sleepsonthefloor**

### CODE REVIEW

File: ci/docker/os-image-stemcell-builder/scripts/update.sh [hunk 1]
```diff
@@ -69,6 +69,9 @@ apt-get -y install qemu-utils
 # CentOS building requirements
 apt-get -y install yum
 
+# Oracle image building requirements
+apt-get -y dosfstools gdisk
```
Comment (dmutreja) [inline ci/docker/os-image-stemcell-builder/scripts/update.sh]: Missing command install.  Should be `apt-get -y install dosfstools gdisk`

Comment (sleepsonthefloor) [reply ci/docker/os-image-stemcell-builder/scripts/update.sh]: nice catch. I have not had a chance to try the image build yet actually - I'll give it a go once our office move settles a bit.

File: bosh-stemcell/spec/stemcells/stig_spec.rb [hunk 2]
```diff
@@ -157,6 +157,7 @@
       Bosh::Stemcell::Arch.ppc64le? &&
         ['V-38579', 'V-38581', 'V-38583', 'V-38585'].include?(stig)
     end
+
```
Comment (vishvananda) [inline bosh-stemcell/spec/stemcells/stig_spec.rb]: did you intend to add this blank line?

File: bosh-stemcell/spec/stemcells/ubuntu_trusty_spec.rb [hunk 3]
```diff
@@ -282,15 +293,29 @@
...
+    } do
+      it { should be_file }
+      its (:content) { should eq("LABEL=cloudimg-rootfs\t/\t ext4\tdefaults\t0 0\nLABEL=EFI\t/boot/efi\tvfat\tdefaults\t0 0\n") }
+    end
   end
 
   describe 'installed packages' do
     dpkg_list_packages = "dpkg --get-selections | cut -f1 | sed -E 's/(linux.*4.4).*/\\1/'"
 
+    let(:dpkg_list_oracle_ubuntu) { File.read(spec_asset('dpkg-list-oracle-ubuntu.txt')) }
```
Comment (vishvananda) [inline bosh-stemcell/spec/stemcells/ubuntu_trusty_spec.rb]: probably switch to readlines and chop like below

Comment (vishvananda) [reply bosh-stemcell/spec/stemcells/ubuntu_trusty_spec.rb]: also probably switch to using the same base.

File: stemcell_builder/stages/system_oracle/assets/etc/cloud/cloud.cfg [hunk 4]
```diff
@@ -0,0 +1,111 @@
+# The top level settings are used as module
```
Comment (vishvananda) [inline stemcell_builder/stages/system_oracle/assets/etc/cloud/cloud.cfg:1]: probably should base this off of the cloud.cfg from our existing ubuntu cloud image.

**Pull request merged**